wordpress-mobile/WordPress-Android
# Add day selection screen - part 1

Fixes #14705 

This PR adds functionality to select days for blogging reminders. This is only the first part of the UI changes because the code changes started to grow too much. The functionality to select a day for a reminder is there but it doesn't trigger a notification yet. 

What's missing
- polish the UI - especially fix the Diff utils for the day buttons so they don't all get refreshed when one is selected and the margins. Let's leave the design review for the second part.
- add the missing views under the Day buttons block

To test:
- Make sure the Blogging reminders flag is turned on
- Go to My Site/Settings
- Click on Blogging goals
- Notice the new updated UX
- Click on a few days 
- Confirm selection
- Notice the summary of the settings item is updated
- Click on a few Blogging goals
- Close bottom sheet
- Notice the changes are discarded and the summary is not updated


To test: 
- Turn on the app with "US English" language
- Notice the Days start with Sunday
- Change locale to some other language that starts week with Monday (for example Czech)
- Notice the Days start with Monday
- Check the same days as before are selected - only the days are shifted

<img width="501" alt="Screenshot 2021-06-02 at 18 11 13" src="https://user-images.githubusercontent.com/1079756/120515337-819a8980-c3ce-11eb-9c24-4c81d3e5f219.png">


## Regression Notes
1. Potential unintended areas of impact
None

PR submission checklist:
